### PR TITLE
WFLY-15555 - Move WildFly Preview to native Jakarta namespace variants of the opentelemetry subsystem modules - 

### PR DESCRIPTION
**Pull request by jasondlee**
https://issues.redhat.com/browse/WFLY-15555

Add source transform modules

### CODE REVIEW

File: ee-9/source-transform/observability/opentelemetry-api/pom.xml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+            <exclusions><exclusion><groupId>*</groupId><artifactId>*</artifactId></exclusion></exclusions>
+        </dependency>
+        <dependency>
+            <groupId>org.wildfly.security</groupId>
+            <artifactId>wildfly-elytron-security-manager</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>jakarta.annotation</groupId>
+            <artifactId>jakarta.annotation-api</artifactId>
+            <version>2.0.0</version>
```
Comment (bstansberry) [inline ee-9/source-transform/observability/opentelemetry-api/pom.xml:89]: This isn't needed.

File: ee-9/source-transform/observability/opentelemetry/pom.xml [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+        <transformer-input-dir>${project.basedir}/../../../../observability/observability</transformer-input-dir>
+    </properties>
+
+    <dependencies>
+        <!-- Jakarata-native Deps -->
+
+        <dependency>
+            <groupId>org.jboss.spec.javax.ws.rs</groupId>
+            <artifactId>jboss-jaxrs-api_3.0_spec</artifactId>
+            <version>${version.org.jboss.spec.jakarta.ws.rs.jboss-jaxrs-api_3.0_spec}</version>
```
Comment (bstansberry) [inline ee-9/source-transform/observability/opentelemetry/pom.xml:54]: Shouldn't be needed.

File: ee-9/source-transform/observability/opentelemetry/pom.xml [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+        </dependency>
+
+        <!-- OpenTelemetry Deps -->
+        <dependency>
+            <groupId>${project.groupId}</groupId>
+            <artifactId>wildfly-opentelemetry-api</artifactId>
```
Comment (bstansberry) [inline ee-9/source-transform/observability/opentelemetry/pom.xml:60]: I'd expect this to be wildfly-opentelemetry-api-jakarta

File: ee-9/source-transform/observability/opentelemetry/pom.xml [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+            <version>${project.version}</version>
```
Comment (bstansberry) [inline ee-9/source-transform/observability/opentelemetry/pom.xml:61]: Shouldn't be needed.

File: ee-9/source-transform/observability/opentelemetry/pom.xml [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+
+        <dependency>
+            <groupId>io.opentelemetry</groupId>
+            <artifactId>opentelemetry-api</artifactId>
+            <exclusions>
+                <exclusion>
+                    <groupId>*</groupId>
+                    <artifactId>*</artifactId>
+                </exclusion>
+            </exclusions>
```
Comment (bstansberry) [inline ee-9/source-transform/observability/opentelemetry/pom.xml:72]: At a glance it seems like all these io.opentelemetry deps have these exclusions in the dependencyManagement section of the root pom. So let's not repeat them here.

Same applies to any place in this PR where things from dependencyManagement (incl ee-9/pom.xml) are being duplicated. opentelemetry-pai/pom.xml has a number of these.

File: ee-9/source-transform/observability/opentelemetry/pom.xml [hunk 6]
```diff
@@ -0,0 +1,266 @@
...
+        </dependency>
+        <dependency>
+            <groupId>${ee.maven.groupId}</groupId>
+            <artifactId>wildfly-ee</artifactId>
```
Comment (bstansberry) [inline ee-9/source-transform/observability/opentelemetry/pom.xml:180]: This should be wildfly-ee-jakarta

File: ee-9/source-transform/observability/opentelemetry/pom.xml [hunk 7]
```diff
@@ -0,0 +1,266 @@
...
+                <exclusion>
+                    <groupId>*</groupId>
+                    <artifactId>*</artifactId>
+                </exclusion>
+            </exclusions>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>${ee.maven.groupId}</groupId>
+            <artifactId>wildfly-weld-common</artifactId>
```
Comment (bstansberry) [inline ee-9/source-transform/observability/opentelemetry/pom.xml:191]: wildfly-weld-common-jakarta

File: ee-9/source-transform/observability/opentelemetry/pom.xml [hunk 8]
```diff
@@ -0,0 +1,266 @@
...
+            <groupId>org.jboss.resteasy</groupId>
+            <artifactId>resteasy-core-spi</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.jboss.metadata</groupId>
+            <artifactId>jboss-metadata-web</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>${project.groupId}</groupId>
+            <artifactId>wildfly-web-common</artifactId>
```
Comment (bstansberry) [inline ee-9/source-transform/observability/opentelemetry/pom.xml:258]: wildfly-eb-common-jakarta

File: ee-9/source-transform/observability/opentelemetry/pom.xml [hunk 9]
```diff
@@ -0,0 +1,266 @@
...
+        </dependency>
+        <dependency>
+            <groupId>${project.groupId}</groupId>
+            <artifactId>wildfly-jaxrs</artifactId>
+        </dependency>
```
Comment (bstansberry) [inline ee-9/source-transform/observability/opentelemetry/pom.xml:263]: I'm not sure why this is here, and why it's in the non-ee9 subsystem pom either. Same goes for a number of the deps here (and there).

File: ee-9/feature-pack/pom.xml [hunk 10]
```diff
@@ -517,6 +517,17 @@
...
+                
+                <exclusion>
+                    <groupId>${ee.maven.groupId}</groupId>
+                    <artifactId>wildfly-opentelemetry-api</artifactId>
+                </exclusion>
+
+                <exclusion>
+                    <groupId>${ee.maven.groupId}</groupId>
+                    <artifactId>wildfly-opentelemetry</artifactId>
+                </exclusion>
```
Comment (bstansberry) [inline ee-9/feature-pack/pom.xml:529]: Please move these two below, in the right spot alphabetically with the others.

File: ee-9/pom.xml [hunk 11]
```diff
@@ -889,6 +889,30 @@
...
+                <groupId>${ee.maven.groupId}</groupId>
+                <artifactId>wildfly-opentelemetry-jakarta</artifactId>
+                <version>${ee.maven.version}</version>
+                <exclusions>
+                    <exclusion>
+                        <groupId>*</groupId>
+                        <artifactId>*</artifactId>
+                    </exclusion>
+                </exclusions>
+            </dependency>
```
Comment (bstansberry) [inline ee-9/pom.xml:914]: Please move these two up in their alphabetical position.

I'm being a bit anal about these poms. :)

File: ee-9/source-transform/observability/opentelemetry/pom.xml (hunk 2 above)
Comment (bstansberry) [reply ee-9/source-transform/observability/opentelemetry/pom.xml:54]: To clarify: all these 'Shouldn't be needed' comments are referring to the specific xml line, i.e. the version element, not the entire dependency block.

**Review by bstansberry — COMMENTED** [review]
I've made a bunch of comments, but in general this looks right. It looks like it failed CI (I haven't looked why).

File: ee-9/source-transform/observability/opentelemetry/pom.xml (hunk 9 above)
Comment (jasondlee) [reply ee-9/source-transform/observability/opentelemetry/pom.xml:263]: I can't say offhand, but I like to think there was a reason, deduced some time ago, that I just don't remember. :) Once I get things working, I'll try verifying the need for some of these to clean things up.

File: ee-9/feature-pack/pom.xml (hunk 10 above)
Comment (jasondlee) [reply ee-9/feature-pack/pom.xml:529]: lol. Wow. I thought I had. :P Moved.

File: ee-9/source-transform/observability/opentelemetry/pom.xml [hunk 12]
```diff
@@ -0,0 +1,186 @@
...
+    </parent>
+
+    <artifactId>wildfly-opentelemetry-jakarta</artifactId>
+
+    <name>WildFly: OpenTelemetry (Jakarta Namespace)</name>
+	
+    <packaging>jar</packaging>
+
+    <properties>
+        <transformer-input-dir>${project.basedir}/../../../../observability/observability</transformer-input-dir>
```
Comment (bstansberry) [inline ee-9/source-transform/observability/opentelemetry/pom.xml:45]: The second 'observability' should be 'opentelemetry'

File: ee-9/source-transform/observability/opentelemetry-api/pom.xml [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+    </parent>
+
+    <artifactId>wildfly-opentelemetry-api-jakarta</artifactId>
+
+    <name>WildFly: OpenTelemetry API (Jakarta Namespace)</name>
+	
+    <packaging>jar</packaging>
+
+    <properties>
+        <transformer-input-dir>${project.basedir}/../../../../observability/observability-api</transformer-input-dir>
```
Comment (bstansberry) [inline ee-9/source-transform/observability/opentelemetry-api/pom.xml:45]: The second 'observability-api' should be 'opentelemetry-api'

**Review by bstansberry — APPROVED** [review]

**Pull request merged**